kythe/go/platform/tools/kzip_validator/kzip_validator.go

```diff
@@ -109,16 +152,123 @@ func main() {
 			fmt.Printf("%10d %s\n", v.Missing, v.Path)
 		}
 	}
+	return
+}
+
+func nocleanup() error {
+	return nil
 }
 
 // fetchRepo either just early returns an available -local_repo, or fetches it from
 // the specified remote -repo_url
-func fetchRepo() string {
+func fetchRepo() (string, func() error, error) {
```

Comment:
Simplify to just working with a remote repository: `func fetchRepo(r *remoteRepoConfig) (path string, cleanup func() error, err error) { ... }`